### PR TITLE
Update kubernetes SD documentation

### LINKED ISSUE #455: Clarify podspec requirements for Kubernetes pod/container discovery
As discussed in prometheus/prometheus#1693, the pod discovery requires named ports. This is kind of implied by

> For pods, the port is the "first port" for the first container reported by Kubernetes.

However I believe this is also incorrect since @pdbogen in the end changed prometheus/prometheus#1449 to look at [all containers](https://github.com/prometheus/prometheus/blob/e3ec8fa83bad9ebaaad608af10d7c9f1416332a6/retrieval/discovery/kubernetes/discovery.go#L934).

I'm not quite sure whether _naming_ the ports is actually required, or will this default to empty?

### PR DESCRIPTION
**Pull request by pdbogen**
closes #455

### CODE REVIEW

**Comment by matthiasr** [comment]
👍

**Comment by matthiasr** [comment]
Thank you for the quick response!

**Comment by pdbogen** [comment]
No problem. I know how frustrating it can be for a user when the docs are incomplete or incorrect. :)

File: content/docs/operating/configuration.md [hunk 1]
```diff
@@ -317,20 +317,29 @@ create multiple target groups:
 
 * One for all API servers, with each API server as a target
 (`__meta_kubernetes_role=apiserver`)
+    * The default port is that used for the API server itself.
```
Comment (beorn7) [inline content/docs/operating/configuration.md]: If you s/that/the port/, it reads easier to me.

File: content/docs/operating/configuration.md [hunk 2]
```diff
@@ -317,20 +317,29 @@ create multiple target groups:
...
 * One for all nodes, with each node as a target (`__meta_kubernetes_role=node`)
+    * The default port is the Kubelet Port from the Kubernetes configuration.
```
Comment (beorn7) [inline content/docs/operating/configuration.md]: Is there a reason to capitalize "Port"?

File: content/docs/operating/configuration.md [hunk 3]
```diff
@@ -317,20 +317,29 @@ create multiple target groups:
...
 * One for all services, with each service as a target
 (`__meta_kubernetes_role=service`)
+    * The default port is selected in an undefined manner. If you have multiple
+    ports on your service, you should explicitly configure relabeling to
```
Comment (beorn7) [inline content/docs/operating/configuration.md]: s/on/for/ ?

File: content/docs/operating/configuration.md [hunk 4]
```diff
@@ -317,20 +317,29 @@ create multiple target groups:
...
+    indicate the proper port.
 * One for *each* service, with each service endpoint as a target
 (`__meta_kubernetes_role=endpoint`)
+    * The default port is selected in an undefined manner. If you have multiple
+    ports on your service, you should explicitly configure relabeling to
```
Comment (beorn7) [inline content/docs/operating/configuration.md]: Same.

File: content/docs/operating/configuration.md [hunk 5]
```diff
@@ -317,20 +317,29 @@ create multiple target groups:
...
+    indicate the proper port.
 * One for all pods, with each pod as a target (`__meta_kubernetes_role=pod`)
+    * The default port is the numerically lowest port on the container that is
```
Comment (beorn7) [inline content/docs/operating/configuration.md]: Perhaps s/on/exposed by/ ? I seem to not like the "on" in connection with ports. :)

**Comment by beorn7** [comment]
Thanks. Looks great.

Just a few nitpicking suggestions above...

**Comment by pdbogen** [comment]
@beorn7 I believe the changes address your comments; though I mostly didn't make _precisely_ the changes you suggested. Let me know if you see anything that could be improved.

**Pull request merged**